DataManagementSystem/Service/FileCatalogHandler.py

```diff
@@ -523,16 +535,41 @@ def export_removeMetadata(self, pathMetadataDict):
   types_getDirectoryUserMetadata = [StringTypes]
 
   def export_getDirectoryUserMetadata(self, path):
-    """ Get all the metadata valid for the given directory path
-    """
-    return gFileCatalogDB.dmeta.getDirectoryMetadata(path, self.getRemoteCredentials())
+    """ Get all the metadata valid for the given directory path.
+        Client gets metadata w/o the groupr suffix.
+    """
+    metaDict = gFileCatalogDB.dmeta.getDirectoryMetadata(path, self.getRemoteCredentials())
+    gLogger.debug("Directory metadata RPC",metaDict)
+    # code below breaks client-side checks
+    suffix = Utilities.fqMetaNameSuffix(self.getRemoteCredentials())
+    if metaDict['OK']:
+      if metaDict['Value']:
+        metaDict['Value'] = {key.replace(suffix,'') if key.endswith(suffix)
+                             else key:value for key, value in metaDict['Value'].iteritems()}
+      if metaDict['MetadataOwner']:
+        metaDict['MetadataOwner']= {key.replace(suffix,'') if key.endswith(suffix)
+                             else key:value for key, value in metaDict['Value'].iteritems()}
```

Comment:
Should it be "else key:value for key, value in metaDict['MetadataOwner'].iteritems()}" ?

Reply:
Good spot !